### PR TITLE
adds error message and prevents user from creating objects and frames…

### PR DESCRIPTION
**Pull request by benptc**
… with non-alphanumeric names

### CODE REVIEW

**Comment by benptc** [comment]
Shows an error and prevents creation when Add button pressed e.g:

Screen Shot 2020-04-14 at 2 41 09 PM

File: libraries/webInterface/gui/index.js [hunk 1]
```diff
@@ -1571,6 +1568,27 @@ realityServer.gotClick = function (event) {
...
+function isNameOk(textContent) {
+    if (textContent === 'Enter Name') {
+        return false;
+    }
+    if (textContent === '') {
+        return false;
+    }
+    let isAlphanumeric = /^[a-z0-9]+$/i.test(textContent);
+    if (!isAlphanumeric) {
+        showErrorNotification('Name must be alphanumeric');
```
Comment (hobinjk-ptc) [inline libraries/webInterface/gui/index.js:1587]: It must be lowercase as well according to the regex

File: libraries/webInterface/gui/index.js [hunk 2]
```diff
@@ -1429,88 +1429,85 @@ realityServer.gotClick = function (event) {
...
-                if (shouldAddWorldObject) {
-                    objectName = '_WORLD_' + textContent;
-                }
+        let objectName = textContent;
+        if (shouldAddWorldObject) {
+            objectName = '_WORLD_' + textContent;
+        }
 
-                // TODO: sanitize object names before creating, for example prevent _WORLD_local (github issue #20)
+        // TODO: sanitize object names before creating, for example prevent _WORLD_local (github issue #20)
```
Comment (hobinjk-ptc) [inline libraries/webInterface/gui/index.js:1445]: This is handled by isNameOk as far as I can see

File: libraries/webInterface/gui/index.js (hunk 1 above)
Comment (benptc) [reply libraries/webInterface/gui/index.js:1587]: oops I'll fix the regex to not be lowercase, I think mixed case should be allowed

File: libraries/webInterface/gui/index.js (hunk 2 above)
Comment (benptc) [reply libraries/webInterface/gui/index.js:1445]: ah yes, being able to not create non-world objects with the same name as world objects is a non-intended consequence of this

**Comment by benptc** [comment]
Fixes issue #20

**Pull request merged**